ui-library-click-to-call/README.md

```diff
@@ -47,3 +53,1211 @@ Instead, it will copy all the configuration files and the transitive dependencie
+            window.origin + `/?${startNewSessionString}`,
+            'call screen',
+            'width=500, height=450'
+        );
+    }, []);
+    ...
+```
+
+What this handler does is start a new window position and place a new query arg in the window URL so that the main application knows that it is time to start a new call.
+
+Next we will add a `useEffect` hook that is creating a event handler listening for new post messages from the child window.
+
+`ClickToCallScreen.tsx`
+```typescript
+    ...
+    useEffect(() => {
+        window.addEventListener('message', (event) => {
+            if (event.origin !== window.origin) {
+                return;
+            }
+            if (event.data === 'args please') {
+                const data = {
+                    userId: adapterParams.userId,
+                    displayName: adapterParams.displayName,
+                    token: adapterParams.token,
+                    locator: adapterParams.locator,
+                    alternateCallerId: adapterParams.alternateCallerId,
+                    useVideo: useVideo
+                };
+                console.log(data);
+                newWindowRef.current?.postMessage(data, window.origin);
+            }
+        });
+    }, [adapterParams, adapterParams.locator, adapterParams.displayName, useVideo]);
+    ...
+```
+
+This handler will listen for events from the child window, (**NOTE: make sure that if the origin of the message is not from your app then return**) if the child window asks for arguments we will send it the arguments needed to construct a `AzureCommunicationsCallAdapter`.
+
+Finally on this screen we will add the `startNewWindow` handler to the widget so that it knows to create the new window.
+
+```typescript
+    ...
+    <Stack horizontal tokens={{ childrenGap: '1.5rem' }} style={{ overflow: 'hidden', margin: 'auto' }}>
+        <ClickToCallComponent
+            onRenderStartCall={startNewWindow}
+            onRenderLogo={() => {
+                return (
+                    <img
+                        style={{ height: '4rem', width: '4rem', margin: 'auto' }}
+                        src={hero}
+                        alt="logo"
+                    />
+                );
+            }}
+            onSetDisplayName={setUserDisplayName}
+            onSetUseVideo={setUseVideo}
+        />
+    </Stack>
+    ...
+```
+
+Next we will need to update the `App.tsx` file to listen for and ask for the messages from what would be the parent window. First to start, you might recall that we added a new query parameter to the URL of the application `newSession=true`. To use this and have our app look for that in the URL we will need to create a utility function to parse out that parameter so our application will behave differently when it is received.
+
+To do that lets add a new folder `src/utils` and in this folder we will add the file `AppUtils.ts`. In this file we will put the following function:
```

Comment:
Reworded so that "we will need to update App.tsx" is not the first thing we read. We should read that we will be modifying the src/utils/AppUtils.tsx file next inorder to have App.tsx do it's thing.